### PR TITLE
Add logs to Dapr Workflows

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

This PR does the following:
1. Adds the support for adding logger in Dapr
2. Adds logs in dapr-ext-workflow
3. Passes the logger options to DTF-python to enable logging

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #626 
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by berndverst — CHANGES_REQUESTED** [review]
Unless you are ready to refactor all other logging code for the rest of the Python SDKs please move the logger into a sub package of `dapr-ext-workflow` for now. (`ext/dapr-ext-workflow` sub folder)

**Comment by berndverst** [comment]
My strong preference is for self contained PRs. So if you'd like to introduce a logger to be used for all Python SDKs, I would like to see that done in a single PR. That's why I am asking for the logger code to be moved to the workflow extension directory for now. As long as the code is self contained there I will approve :)

In the future we could still migrate that to be at the top level of the repo, but at that point I'd like the same PR to contain the switch to that logger throughout all SDKs.

**Comment by shivamkm07** [comment]
> My strong preference is for self contained PRs. So if you'd like to introduce a logger to be used for all Python SDKs, I would like to see that done in a single PR. That's why I am asking for the logger code to be moved to the workflow extension directory for now. As long as the code is self contained there I will approve :)
> 
> In the future we could still migrate that to be at the top level of the repo, but at that point I'd like the same PR to contain the switch to that logger throughout all SDKs.

Done

File: dapr/logger/options.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from typing import Union
+import logging
+
+
+class LoggerOptions:
```
[the remark this answers is not in the document]
Comment (shivamkm07) [reply dapr/logger/options.py]: @mukundansundar Added logger class. Please review

File: ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py [hunk 2]
```diff
@@ -49,26 +55,33 @@ def is_replaying(self) -> bool:
...
         def wf(ctx: task.OrchestrationContext, inp: TInput):
-            daprWfContext = DaprWorkflowContext(ctx)
+            daprWfContext = DaprWorkflowContext(ctx, self._logger.get_options())
             return workflow(daprWfContext, inp)
         # copy workflow name so durabletask.worker can find the orchestrator in its registry
         wf.__name__ = workflow.__name__
         return self.__obj.call_sub_orchestrator(wf, input=input, instance_id=instance_id)
 
     def wait_for_external_event(self, name: str) -> task.Task:
+        self._logger.debug(f'instance_id {self.instance_id} wait_for_external_event: {name}')
```
Comment (mukundansundar) [inline ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py:80]: if we are going to follow a k:v approach, I would prefer to use a utility method with a particular format. The other option is we have full sentences in logs similar to the other logs in python-sdk.

File: ext/dapr-ext-workflow/dapr/ext/workflow/logger/options.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        # Set default log level to INFO if none is provided
+        if log_level is None:
+            log_level = logging.INFO
+        # Add a default log handler if none is provided
+        if log_handler is None:
+            log_handler = logging.StreamHandler()
+        # Set a default log formatter if none is provided
+        if log_formatter is None:
+            log_formatter = logging.Formatter(
+                fmt="%(asctime)s.%(msecs)03d %(name)s %(levelname)s: %(message)s",
```
Comment (mukundansundar) [inline ext/dapr-ext-workflow/dapr/ext/workflow/logger/options.py:36]: is this format similar to the default format for python logs? 

If not, is there a possibility that when using workflow SDK, there will be two different format of logs from the SDK?

**Review by mukundansundar — COMMENTED** [review]
overall lgtm

File: ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py (hunk 2 above)
Comment (shivamkm07) [reply ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py:80]: The formatter is added as part of `LoggerOptions`. It's set to `TIME SCOPE LOG_LEVEL: LOG_MESSAGE` by default.

File: ext/dapr-ext-workflow/dapr/ext/workflow/logger/options.py (hunk 3 above)
Comment (shivamkm07) [reply ext/dapr-ext-workflow/dapr/ext/workflow/logger/options.py:36]: For now, there is no logging capability in python-sdk yet

File: ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py (hunk 2 above)
Comment (mukundansundar) [reply ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py:80]: but then you are formatting the message as `k:v` pairs ... if so would suggest to add a simple utility method for that

**Review by mukundansundar — APPROVED** [review]

File: ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py (hunk 2 above)
Comment (shivamkm07) [reply ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py]: Moved to sentences

**Review by mukundansundar — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**